src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,130 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private final double defaultLoadFactor = 0.75;
+    private int threshold = (int) (DEFAULT_INITIAL_CAPACITY * defaultLoadFactor);
+    private Entry<K, V>[] buckets;
+    private List<Entry<K, V>> nullBucket;
+    private int size;
 
-    @Override
-    public void put(K key, V value) {
+    private static class Entry<K, V> {
+        private final K key;
+        private V value;
+        private Entry<K, V> next;
+
+        public Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        void setValue(V value) {
+            this.value = value;
+        }
+    }
 
+    public MyHashMap() {
+        buckets = new Entry[DEFAULT_INITIAL_CAPACITY];
+        nullBucket = new ArrayList<>();
+    }
+
+    public MyHashMap(int initialCapacity) {
+        this.threshold = (int) (initialCapacity * defaultLoadFactor);
+        this.buckets = new Entry[initialCapacity];
+        nullBucket = new ArrayList<>();
+    }
+
+    private void resize() {
+        int newCapacity = buckets.length * 2;
```

Comment:
magic number, let's extract to constant GROW_FACTOR which will improve readability